sbndaq-artdaq/Generators/ICARUS/PhysCrateData_generator.cc

```diff
@@ -104,9 +104,23 @@ void icarus::PhysCrateData::SetCompressionBits()
   TRACEN("PhysCrateData", TLVL_INFO, "Enabled TPC hardware compression scheme %ld", _compressionScheme);
 }
 
-// read the firmware version off the A2795 boards
+// read the firmware versions
 void icarus::PhysCrateData::ReadFirmwareVersion()
 {
+  // A3818 firmware / driver
+  for (const auto& link : pcieLinks_)
+  {
+    int32_t bdhandle;
+    CAENVME_Init2(cvA3818, &link, 0, &bdhandle);
```

Comment:
This line basically opens a new connection, and we need to verify it doesn't create conflicts. The tests I did with the V1730s showed no issues, but it might also depend when you're calling this function.